tests/virtctl/vnc.go

```diff
@@ -51,129 +49,73 @@ import (
-				Expect(err).ToNot(HaveOccurred())
-				return img
-			}, 10*time.Second).Should(HaveResolution(xres, yres))
-		})
+		}()
+
+		Eventually(func(g Gomega) {
+			conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%s", testPort))
+			g.Expect(err).ToNot(HaveOccurred())
+			defer conn.Close()
+
+			client, err := vnc.Client(conn, &vnc.ClientConfig{
+				ServerMessageCh: make(chan vnc.ServerMessage),
+				ServerMessages:  []vnc.ServerMessage{new(vnc.FramebufferUpdateMessage)},
+			})
+			g.Expect(err).ToNot(HaveOccurred())
+			defer client.Close()
+
+			g.Expect(client.DesktopName).To(ContainSubstring(vmi.Name))
+		}, 60*time.Second).Should(Succeed())
 	})
-})
 
-func getResolution(domain *launcherApi.DomainSpec) (X, Y int) {
-	videoType := domain.Devices.Video[0].Model.Type
-	if videoType == "virtio" {
-		X = 1280
-		Y = 800
-	} else {
-		X = 720
-		Y = 400
-	}
-	return X, Y
-}
-
-type ResolutionMatcher struct {
-	X, Y int
-}
-
-func (h ResolutionMatcher) Match(actual interface{}) (success bool, err error) {
-	x, y, err := imgSize(actual)
-	if err != nil {
-		return false, nil
-	}
-	return x == h.X && y == h.Y, nil
-}
-
-func (h ResolutionMatcher) FailureMessage(actual interface{}) (message string) {
-	x, y, err := imgSize(actual)
-	if err != nil {
-		return err.Error()
-	}
-	return fmt.Sprintf("Expected (X: %d, Y: %d) to match (X: %d, Y: %d)", x, y, h.X, h.Y)
-}
-
-func (h ResolutionMatcher) NegatedFailureMessage(actual interface{}) (message string) {
-	x, y, err := imgSize(actual)
-	if err != nil {
-		return err.Error()
-	}
-	return fmt.Sprintf("Expected (X: %d, Y: %d) to not match (X: %d, Y: %d)", x, y, h.X, h.Y)
-}
-
-func imgSize(actual interface{}) (X, Y int, err error) {
-	if actual == nil {
-		return -1, -1, fmt.Errorf("expected an object of type image.Image but got nil")
-	}
-	img, ok := actual.(image.Image)
-	if !ok {
-		return -1, -1, fmt.Errorf("expected an object of type image.Image")
-	}
-	size := img.Bounds().Size()
-	return size.X, size.Y, nil
-}
-
-func HaveResolution(X, Y int) ResolutionMatcher {
-	return ResolutionMatcher{X: X, Y: Y}
-}
+	It("should allow creating a VNC screenshot in PNG format", func() {
+		domain, err := tests.GetRunningVMIDomainSpec(vmi)
+		Expect(err).ToNot(HaveOccurred())
+		// The default resolution is 720x400 for vga device while it is 1280x800 for virtio-gpu device
+		size := image.Point{720, 400}
+		if domain.Devices.Video[0].Model.Type == "virtio" {
+			size = image.Point{1280, 800}
+		}
+
+		By("gathering screenshots until we are past the first boot screen and see the expected resolution")
+		path := filepath.Join(GinkgoT().TempDir(), "screenshot.png")
+		Eventually(func(g Gomega) image.Point {
+			err := clientcmd.NewRepeatableVirtctlCommand(
+				"vnc", "screenshot",
+				vmi.Name,
+				"--namespace", vmi.Namespace,
+				"--file="+path,
+			)()
+			g.Expect(err).ToNot(HaveOccurred())
+
+			file, err := os.Open(path)
+			g.Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(file.Close)
```

Comment:
Here you should use defer so ever cycle would close it, imho

Reply:
ACK